### PR TITLE
assisted-service: Added release branch config

### PR DESCRIPTION
**Pull request by sagidayan**
Added config and generated jobs for assisted-service release branches.

Jira: [MGMT-6421](https://issues.redhat.com/browse/MGMT-6421)

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    namespace: ocp-kni
+    tag: dev-scripts
+  openshift_release_golang-1.15:
+    name: release
+    namespace: openshift
+    tag: golang-1.15
+  openshift_release_golang-1.16:
+    name: release
+    namespace: openshift
+    tag: golang-1.16
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:37]: it's weird we need them both
@YuviGold / @lranjbar why we have those both on the assisted-service jobs? (for both master and ocm-2.3)
anyway, not a blocker for this PR

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  inputs:
+    openshift_release_golang-1.16:
+      as:
+      - registry.ci.openshift.org/openshift/release:golang-1.16
+  to: assisted-service
+- dockerfile_literal: |
+    FROM coreos/centos
+    COPY . .
+  from: src
+  to: assisted-service-scripts
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:59]: 👍🏽

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  - pullspec: quay.io/ocpmetal/assisted-service:latest
+    with: assisted-service
+  - pullspec: quay.io/ocpmetal/assisted-installer:latest
+    with: assisted-installer
+  - pullspec: quay.io/ocpmetal/assisted-installer-agent:latest
+    with: assisted-installer-agent
+  - pullspec: quay.io/ocpmetal/assisted-installer-controller:latest
+    with: assisted-installer-controller
+  - pullspec: quay.io/ocpmetal/postgresql-12-centos7:latest
+    with: postgresql-12-centos7
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:73]: 👍🏽

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+releases:
+  candidate:
+    release:
+      channel: candidate
+      version: "4.8"
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:59]: 👍🏽

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml (hunk 1 above)
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:37]: @osherdp  Ill leave these in. If it is not needed, We can remove them from all needed branches together. Dont wwant to add other configs in this PR.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    namespace: coreos
+    tag: latest
+  coreos_centos_stream8:
+    name: stream8
+    namespace: coreos
+    tag: "8"
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
```
[the remark this answers is not in the document]
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml]: 👍🏽

**Comment by osherdp** [comment]
/hold
(just to see the results of rehearse jobs which are optional. unhold when it's done)

**Review by osherdp — APPROVED** [review]

**Comment by YuviGold** [comment]
@gamli75 I think the name of the branch should be without `hotfix` as it should be the actual cloud release branch.
IMHO either `release` or `cloud_release`. WDYT?

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (sagidayan) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml]: Seems like `dev-scripts` are needed for packet tests.

**Review by osherdp — APPROVED** [review]

**Comment by gamli75** [comment]
> @gamli75 I think the name of the branch should be without `hotfix` as it should be the actual cloud release branch.
> IMHO either (`release`, `cloud`, or `cloud-release`). WDYT?

it depends on where I'm releasing from - most of the time I'm releasing from the master branch but if QE will find a critical issue - I'll release it from the hotfix branch.

**Pull request merged**